test/jdk/jdk/internal/reflect/CallerSensitive/CallerSensitiveFinder.java

```diff
@@ -87,75 +91,83 @@ public static void main(String[] args) throws Exception {
     }
 
     private final List<String> csMethodsMissingAnnotation = new CopyOnWriteArrayList<>();
-    private final ReferenceFinder finder;
     public CallerSensitiveFinder() {
-        this.finder = new ReferenceFinder(getFilter(), getVisitor());
         pool = Executors.newFixedThreadPool(numThreads);
 
     }
 
-    private ReferenceFinder.Filter getFilter() {
-        final String classname = "jdk/internal/reflect/Reflection";
-        final String method = "getCallerClass";
-        return new ReferenceFinder.Filter() {
-            public boolean accept(ConstantPool cpool, CPRefInfo cpref) {
-                try {
-                    CONSTANT_NameAndType_info nat = cpref.getNameAndTypeInfo();
-                    return cpref.getClassName().equals(classname) && nat.getName().equals(method);
-                } catch (ConstantPoolException ex) {
-                    throw new RuntimeException(ex);
+    private void check(ClassModel clazz) {
+        final String className = "jdk/internal/reflect/Reflection";
+        final String methodName = "getCallerClass";
+        boolean checkMethods = false;
+        for (var pe : clazz.constantPool()) {
+            if (pe instanceof MethodRefEntry ref
+                    && ref.owner().name().equalsString(className)
+                    && ref.name().equalsString(methodName)) {
+                checkMethods = true;
+            }
+        }
+
+        if (!checkMethods)
+            return;
+
+        for (var method : clazz.methods()) {
+            var code = method.code().orElse(null);
+            if (code == null)
+                continue;
+
+            boolean needsCsm = false;
+            for (var element : code) {
+                if (element instanceof InvokeInstruction invoke
+                        && invoke.opcode() == Opcode.INVOKESTATIC
+                        && invoke.method() instanceof MethodRefEntry ref
+                        && ref.owner().name().equalsString(className)
+                        && ref.name().equalsString(methodName)) {
```

Comment:
Is this additional test necessary, I don't see it in the original test.

Reply:
This was part of the now-removed ReferenceFinder logic; it scans cp as a fast path and then does instruction scan for all methods to find relevant methods to pass to visitor.